lib/jwt/encode.rb

```diff
@@ -51,7 +64,21 @@ def encode_header
     end
 
     def encode_payload
-      encode_data(@payload)
+      # if b64 header is present and false, do not encode payload as per RFC7797 proposed standard
```

Comment:
Would it make sense to not rely on header values but instead have an explicit parameter that controls the behavior of both header and payload encoding?